solc/CommandLineInterface.cpp

```diff
@@ -845,52 +845,56 @@ void CommandLineInterface::handleCombinedJSON()
 	output[g_strVersion] = frontend::VersionString;
 	vector<string> contracts = m_compiler->contractNames();
 
+	// NOTE: The state checks here are more strict that in Standard JSON. There we allow
+	// requesting certain outputs even if compilation fails as long as analysis went ok.
+	bool compilationSuccess = m_compiler->state() >= CompilerStack::State::CompilationSuccessful;
```

Comment:
maybe `bool const`?

Reply:
I don't think it's worth it for simple value types. If I were to do that, I'd be putting `const` on almost every local variable I use. That's not workable unless the language makes variables const by default like Rust.